test/e2e/framework/framework.go

```diff
@@ -210,6 +210,11 @@ func NewFromFlags() (*Framework, error) {
 
 // CreateGameServerAndWaitUntilReady Creates a GameServer and wait for its state to become ready.
 func (f *Framework) CreateGameServerAndWaitUntilReady(ns string, gs *agonesv1.GameServer) (*agonesv1.GameServer, error) {
+	return f.CreateGameServerAndWaitUntilReadyWithOptionalPortCheck(ns, gs, true)
+}
+
+// CreateGameServerAndWaitUntilReadyWithOptionalPortCheck Creates a GameServer and wait for its state to become ready.
+func (f *Framework) CreateGameServerAndWaitUntilReadyWithOptionalPortCheck(ns string, gs *agonesv1.GameServer, portCheck bool) (*agonesv1.GameServer, error) {
```

Comment:
I'm trying to decide if `CreateGameServerAndWaitUntilReadyWithOptionalPortCheck` is necessary.

We already cover this by checking on ports in [this test](https://github.com/googleforgames/agones/blob/170384f993e4919d31950e5f995924de22fec8ef/test/e2e/gameserver_test.go#L54) for example (I'm sure there are others).

@roberthbailey WDYT? 

I get the impetus to check all other tests, but I'm not sure it's required.

Reply:
Looking at your new test code, I'm guessing that you added this because without passing in false, line 227 (in the old file) was returning an error, because there was no port. 

But the test should be able to inspect the passed in game server and tell if a port should be created (by seeing if spec.Ports is nil). And then you can modify the check for ports to read something like "if the spec has a port and none was allocated, return an error" instead of what we have currently ("assume that there will always be a port so check to see if one was allocated").

Reply:
https://github.com/googleforgames/agones/blob/main/test/e2e/framework/framework.go#L227 - just for reference :smile: 

That's a really good point @roberthbailey , that could become:

```go
	if len(readyGs.Status.Ports) == len(gsCopy.Spec.Ports) ​{
		return nil, fmt.Errorf("Ready GameServer instance has no port: %v", readyGs.Status)
	}
```

Don't think we need nil checks either, since `len(nil)` is `0`

Which is far more elegant :+1:

Reply:
I think you mean != 

Also, the error message needs to be updated to print the expected number of ports and the observed number of ports:


```go
	if len(readyGs.Status.Ports) != len(gsCopy.Spec.Ports) ​{
		return nil, fmt.Errorf("Ready GameServer instance has %d port(s), want %v", readyGs.Status, gsCopy.Spec.Ports)
	}
```

Reply:
Ah yes! I did, excellent catch. I like it!

Reply:
Thanks for the input. I've updated the PR accordingly. 👍